### DISCUSSION TITLE
slider to change scalars in add_slider_widget

### DISCUSSION THREAD

**Discussion by venugovh** [pyvista/pyvista#2232] [Q&A]
Hi,

I am trying to add a slider which performs a rolling average operation on a 1D array of scalars with I am plotting for a set of points of the same size as the scalars. The slider should change the kernel size of the rolling average function. The scalar map on the points would vary as I change the kernel size. 

Currently, I have a callback function that takes the inputs - points to plot, the 1D array of scalars, and a kernel_size parameters. But, in the documentation, it is given that the callback function for the slider takes only one input parameter. Is there any way to solve this?
I am not able to figure this out for meshes that are not present in the examples repository of Pyvista.

Any help will be highly appreciated! Thank you so much!

Code snippet!
```
import pyvista as pv
import numpy as np
from scipy.ndimage import uniform_filter1d
points = np.array([[0, 0, 1],[0, 0, 2],[0, 0, 3],[0, 0, 4],
                [0, 0, 5],[0, 0, 6],[0, 0, 7],[0, 0, 8],
                [0, 0, 9],[0, 0, 10],[0, 0, 11],[0, 0, 12]])
scalars = np.array([1,2,3,4,5,6,6,5,4,3,2,1])
p = pv.Plotter()
smooth_scalar1 = uniform_filter1d(scalars, size=12)

def callback_func(scalars, kern_size):
    smooth_scalar = uniform_filter1d(scalars, size=kern_size)
    p.add_points(points, scalars=smooth_scalar, render_points_as_spheres = True)
    return
p.add_points(points, scalars=scalars, render_points_as_spheres = True) # please comment this to test the slider callback function
p.add_points(points, scalars=smooth_scalar1, render_points_as_spheres = True) # please comment this to test the slider callback function
p.add_slider_widget(callback_func, [1, points.shape[0]], title='Smooth Points')
p.show()

```

**Comment by stla** [comment]
Hello,

  Can't you use an array as the argument of the callback function?

**Reply by venugovh** [reply]
The array would change as you vary the `kern_size` parameter. So the slider vary the `kern_size`. I am not sure how passing the array would interact with the slider.

**Comment by venugovh** [comment]
HI @banesullivan ,

Is it possible to return the last slider position in `add_slider_widget`?

Thanks!

**Reply by banesullivan** [reply]
> Is it possible to return the last slider position in add_slider_widget?

I'm not sure I follow what exactly you are asking for

**Comment by banesullivan** [comment]
> Currently, I have a callback function that takes the inputs - points to plot, the 1D array of scalars, and a kernel_size parameters. But, in the documentation, it is given that the callback function for the slider takes only one input parameter. Is there any way to solve this?
I am not able to figure this out for meshes that are not present in the examples repository of Pyvista.

You could try make a class based callable:

```py
class MyCallback:
    def __init__(self, scalars, kern_size)
        self.scalars = scalars
        self.kern_size = kern_size

    def __call__(self, slider_value):
        ... # callback logic using self.kern_size, etc.

...
callback = MyCallback(scalars=..., kern_size=...)
p.add_slider_widget(callback, [1, points.shape[0]], title='Smooth Points')
```
